tidy3d/components/grid/grid_spec.py

```diff
@@ -31,6 +31,7 @@ def make_coords(
         periodic: bool,
         wavelength: pd.PositiveFloat,
         num_pml_layers: Tuple[pd.NonNegativeInt, pd.NonNegativeInt],
+        snapping_points: Tuple[Coordinate, ...],
```

Comment:
consider making this optional `= ()` if it would make sense to minimize changes needed, same with `_make_coords_initial` and the other methods

Reply:
I actually prefer not using optional arguments here: 1) this part is internal, so it doesn't affect user API; 2) when using optional arguments, one often forgets updating codes to pass the new arguments, which can sometimes take a while to debug.

Reply:
sounds fine to me 👍